### ISSUE TITLE
Project Dropdown in Dashboard takes minutes to load

### ISSUE THREAD

**Issue by btzq** [apache/cloudstack#9679]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1.1
~~~

##### SUMMARY

When logging in as Admin, Domain Admin or User, when clicking on the Project Drop down, it doesnt display info at first. 

<img width="484" alt="Screenshot 2024-09-13 at 9 15 21 PM" src="https://github.com/user-attachments/assets/1db55018-12e6-4126-8c9e-1a8afb990e72">

It shows 0 for a long time (around 30s-1min) in order for the project list to be fully loaded. 

But after loading, if you click on the dropdown again, it shows 0 again.

##### STEPS TO REPRODUCE

~~~
1. Login to Cloudstack Dashboard
2. Select 'Project' dropdown
3. Notice it display 0 and No Data
4. Wait for 30s-1min and the project list will be loaded
~~~

##### EXPECTED RESULTS

~~~
Project Dropdown to load faster (eg. 1-3 seconds)
~~~

##### ACTUAL RESULTS

~~~
Project Dropdown takes 30s-1min to load
~~~

**Comment by DaanHoogland** [comment]
@btzq , how many projects and project members do you have?

**Comment by btzq** [comment]
> @btzq , how many projects and project members do you have?

Only around 20 Projects. I dont think its alot. 
As for accounts, you can assume 1 Account for 1 Project.

**Comment by DaanHoogland** [comment]
@btzq I tried to reproduce it but my list filles within a second, of course this is a lab enironment so extra conditions may be needed to reproduce this.  Our dev-lab has 72 projects atm and is very dynamically changing this number. It also has no problems displaying the dropdown within a second in a new private window. The logging on process takes more time so not disturbing at all. I think you'll need to investigate more.

**Comment by btzq** [comment]
@DaanHoogland ive recorded a video of what im seeing. 

https://github.com/user-attachments/assets/c13ad4be-017a-4a35-b6e1-cd7ada099ecb

Notice That:
- Before opening, it shows 0, with a loading icon on the right
- It loads for 45 seconds, even after opening and closing
- When it loads, and i scroll, you can see theres not many projects. 

Strangely, our login process is very fast. No lags there.

Just the Project Dropdown is slow.

**Comment by DaanHoogland** [comment]
> Which version are you using? Could it be you guys are using a newer version that has this issue resolved?

4.19.1.0-SNAPSHOT it is from just before the 4.19.1 release, I'll update it and try again. I do see the same as you but for less than a second.

**Comment by DaanHoogland** [comment]
now tried 4.19.2.0-SNAPSHOT from last night and also it loads almost instantly, sorry @btzq . Can you see any strangities in the console log of the browser?

**Comment by btzq** [comment]
@DaanHoogland does this help?

Note, ive turned off my adblocker just in case. But results still the same. 

<img width="637" alt="Screenshot 2024-09-18 at 9 54 59 PM" src="https://github.com/user-attachments/assets/f109026a-09a2-47d1-ab59-3db2e8971be7">

<img width="624" alt="Screenshot 2024-09-18 at 9 54 08 PM" src="https://github.com/user-attachments/assets/8ce22983-f78c-4273-bbc3-57e7726b57b8">

**Comment by weizhouapache** [comment]
probably a simple fix
```
diff --git a/ui/src/components/header/ProjectMenu.vue b/ui/src/components/header/Proj
+ectMenu.vue
index e2f365344d9..c55db5ff925 100644
--- a/ui/src/components/header/ProjectMenu.vue
+++ b/ui/src/components/header/ProjectMenu.vue
@@ -81,7 +81,7 @@ export default {
       const projects = []
       const getNextPage = () => {
         this.loading = true
-        api('listProjects', { listAll: true, page: page, pageSize: 500, showIcon: tr
+ue }).then(json => {
+        api('listProjects', { listAll: true, page: page, pageSize: 500, showIcon: tr
+ue, details: 'min' }).then(json => {
           if (json?.listprojectsresponse?.project) {
             projects.push(...json.listprojectsresponse.project)
           }
```

**Comment by weizhouapache** [comment]
I will try to reproduce the issue and verify the fix
assigned to myself

**Comment by weizhouapache** [comment]
@btzq 
can you run the following commands on mgmt server if you have downloaded cmk ?

time cmk list projects listall=true filter=id,name >/dev/null
time cmk list projects listall=true filter=id,name details=min >/dev/null

**Comment by btzq** [comment]
@weizhouapache here you go

![WhatsApp Image 2024-09-19 at 11 30 52 PM](https://github.com/user-attachments/assets/389addd5-8760-4d48-9520-f5916aaf7900)

First command take around 36sec
Second command take 0.4sec+

**Comment by weizhouapache** [comment]
> @weizhouapache here you go
> 
> ![WhatsApp Image 2024-09-19 at 11 30 52 PM](https://private-user-images.githubusercontent.com/147712084/369071539-389addd5-8760-4d48-9520-f5916aaf7900.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3NjIxMTksIm5iZiI6MTcyNjc2MTgxOSwicGF0aCI6Ii8xNDc3MTIwODQvMzY5MDcxNTM5LTM4OWFkZGQ1LTg3NjAtNGQ0OC05NTIwLWY1OTE2YWFmNzkwMC5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTlUMTYwMzM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWJmNzhmMWQzZTU5YzZhMmM2OTk5MjQyODFhMTg3ODhmNmVlZTdiNTQ3MzRhOTNiYmQ1YTQ3YTVhMDNmNzBjZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4xcMKs2C7H0v-S8uPVwSWDq_giSJSPho2I3vOLNL73g)
> 
> First command take around 36sec 
> Second command take 0.4sec+

@btzq 
if so, I think #9711 is what you need

**Comment by btzq** [comment]
Awesome, thanks @weizhouapache !

**Comment by DaanHoogland** [comment]
fixed in #9711
